### PR TITLE
Update typeahead select to make it auto select the only option as needed

### PR DESCRIPTION
**Pull request by DaoDaoNoCode**
JIRA: [RHOAIENG-16038](https://issues.redhat.com/browse/RHOAIENG-16038)

## Description

Add a hook in TypeaheadSelect to auto-select the only option and disable the field. This only happens when:
1. The field is required (e.g. When adding a cluster storage, the connected notebook field is not required, we cannot force to select the only workbench)
2. The field is not creatable

## How Has This Been Tested?

Verify all the components that use `TypeaheadSelect` is working as expected.
- If it's creatable, it should not be changed by this PR
- If it's not required, it should not be changed by this PR
- Otherwise, if there is only one option in the select menu list, it should be disabled and auto-select the only option

## Test Impact

Update the cypress test in workbench

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by Gkrumbach07** [comment]
can you update the storage class select. it is disabling the options instead of hiding them

**Comment by DaoDaoNoCode** [comment]
Sure, and I would prefer to hold this PR until https://github.com/opendatahub-io/odh-dashboard/pull/3529 gets merged. I will check the usage with the new connected workbench field.

**Comment by DaoDaoNoCode** [comment]
/hold
There are still some issues with the cluster storage modal. Will check it next week.

**Comment by DaoDaoNoCode** [comment]
/unhold
Should be ready for reviews now.

File: frontend/src/pages/projects/screens/detail/storage/ClusterStorageTableRow.tsx [hunk 1]
```diff
@@ -61,7 +62,7 @@ const ClusterStorageTableRow: React.FC<ClusterStorageTableRowProps> = ({
 
   const selectOptions = availableNotebooks.notebooks.map((notebook) => ({
     value: notebook.metadata.name,
-    content: notebook.metadata.name,
+    content: getDisplayNameFromK8sResource(notebook),
```
Comment (Gkrumbach07) [inline frontend/src/pages/projects/screens/detail/storage/ClusterStorageTableRow.tsx:65]: this is being done here https://github.com/opendatahub-io/odh-dashboard/pull/3584#pullrequestreview-2509549154

**Review by Gkrumbach07 — CHANGES_REQUESTED** [review]

**Review by Gkrumbach07 — APPROVED** [review]
/lgtm

**Pull request merged**